pennylane_lightning/src/tests/Test_KernelMap.cpp

```diff
@@ -0,0 +1,140 @@
+#include "Constant.hpp"
+#include "ConstantUtil.hpp"
+#include "KernelMap.hpp"
+#include "Util.hpp"
+
+#include <catch2/catch.hpp>
+
+using namespace Pennylane;
+using namespace Pennylane::KernelMap;
+
+using Catch::Matchers::Contains;
+
+TEST_CASE("Test PriorityDispatchSet", "[PriorityDispatchSet]") {
+    auto pds = PriorityDispatchSet();
+    pds.emplace(10U, Util::IntegerInterval<size_t>(10, 20),
+                Gates::KernelType::PI);
+
+    SECTION("Test conflict") {
+        /* If two elements has the same priority but integer intervals overlap,
+         * they conflict. */
+        REQUIRE(pds.conflict(10U, Util::IntegerInterval<size_t>(19, 23)));
+    }
+
+    SECTION("Get Kernel") {
+        REQUIRE(pds.getKernel(15) == Gates::KernelType::PI);
+        REQUIRE_THROWS_WITH(pds.getKernel(30),
+                            Contains("Cannot find a kernel"));
```

Comment:
We have a new macro that you can use to REQUIRE_THROWS_AS and REQUIRE_THROWS_WITH in a single line:
`PL_REQUIRE_THROWS_MATCHES(expr, type, message_match)`
With this, you can check for the exception type and match part of the message.
We have a similar one for CHECK:
`PL_CHECK_THROWS_MATCHES(expr, type, message_match)`